OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalRestClient.java

```diff
@@ -103,7 +104,10 @@ private static void makeRequest(final String url, final String method, final JSO
       // If not a GET request, check if the user provided privacy consent if the application is set to require user privacy consent
       if (method != null && OneSignal.shouldLogUserPrivacyConsentErrorMessageForMethodName(null))
          return;
-   
+
+      if (OSUtils.isThreadRunningOnMainThread())
```

Comment:
Let's put this before the privacy consent check. This means we will fail faster so we can detect issues sooner if this become a problem in the future.